### DISCUSSION TITLE
Rename `try .. with ..` to `run .. with ..`

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#9108] [Ideas]
`try` implies that something might not succeed, which is a bit confusing for effects like random. `run .. with ..` has less confusion and avoid mistaking `try .. catch ..` and `run .. with ..` for new people

**Comment by magnus-madsen** [comment]
I like it, but a few things to consider:

- It makes `run` into a keyword, so what do we call `App.run` etc now? Other people might like to use `run` too.
- The research literature uses `try`. I don't think I have seen anything else? Maybe `handle`?
- What will other languages use?

An alternative would be `do`?

**Reply by JonathanStarup** [reply]
another alternative is just to have a infix keyword. `{..} with ..` or `{..} handing ..`

**Reply by magnus-madsen** [reply]
I still think `run` is reasonable though.

**Comment by jaschdoc** [comment]
You could also flip it around and define the handlers first with something like `handle-in`,`fix-in`, `env-in`, etc. (`fix` is already a keyword so maybe not so good?)
```scala
handle FirstEffect {
    ...
} SecondEffect {
    ...
} in f()
```

**Comment by JonathanStarup** [comment]
Another idea from gitter: `do .. with ..`
